deltachat-ios/Chat/ChatViewController.swift

```diff
@@ -382,6 +394,34 @@ class ChatViewController: UITableViewController {
         super.viewDidAppear(animated)
         AppStateRestorer.shared.storeLastActiveChat(chatId: chatId)
 
+        // things that do not affect the chatview
+        // and are delayed after the view is displayed
+        DispatchQueue.global(qos: .background).async { [weak self] in
+            guard let self = self else { return }
+            self.dcContext.marknoticedChat(chatId: self.chatId)
+        }
+
+        handleUserVisibility(isVisible: true)
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+
+        // the navigationController will be used when chatDetail is pushed, so we have to remove that gestureRecognizer
+        navigationController?.navigationBar.removeGestureRecognizer(navBarTap)
+        isDismissing = true
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        isDismissing = false
+        ignoreInputBarChange = true
+        AppStateRestorer.shared.resetLastActiveChat()
+        handleUserVisibility(isVisible: false)
+        audioController.stopAnyOngoingPlaying()
+    }
+
+    private func setupObservers() {
```

Comment:
I moved setting up the observers into an own method, just like it is done for the removal [here](https://github.com/deltachat/deltachat-ios/pull/1358/files#diff-bf4dd3d619f49250044e05ac5777f82c9b4029e49d8d175f6e2bbfefc1fe4685R485)

Reply:
the diff reads more difficult indeed, but the plain code is easier to follow now